docs/user-guide/rules/strict-transport-security.md

```diff
@@ -0,0 +1,123 @@
+# Require `Strict-Transport-Security` response header (`strict-transport-security`)
+
+`strict-transport-security` warns against serving resources over HTTPS without
+`strict-transport-security` header and validates the header directives
+and their corresponding values.
+
+## Why is this important?
+
+Web security has been a crutial concern for the developers nowadays. Unlike
+[cross-site scripting(XSS)][XSS] and [SQL injection](SQL injection), the
+exploit of insufficient protection over the transport layer can be harder to picture
+in practice. If you are a fan of the popular TV show
+"[Silicon Valley][silicon valley]", you might have heard about the intercept of
+wi-fi traffic using a 'pineapple'. It serves as a 'man in the middle' between a
+user's device and the resources on the web. Without being noticed, it secretly inspects
+the entire data flow and steals important user credentials.
+
+Fortunately, this kind of data breach can be avoided to the maximum extent by enforcing
+encrypted connection between the browser and the web resources, i.e., HTTPS. And by
+specifying the `Strict-Transport-Security` header along with a `max-age` value in
+the reponse, the websites declare themselves to be only accessible via the secure
+connections within the specified time period. So for future attempts to the same
+website via the insecure connections, the browswer will automatically redirect
+them back to the secure ones.
+
+Notably, to prevent the `Strict-Transport-Security` header from being stripped by
+the attacker on the user's first visit, major browsers include a "pre-loaded" list
+of sites that must be loaded via HTTPS. After being included in this list, the browswer
+will never be able to make an insecure request to your website, so use with great
+caution - You have to be completely certain that your website, including the
+subdomains, will support HTTPS **forever**.
+
+More information about HTTP Strict Transport (HSTS), please see:
+
+* [HTTP Strict Transport Security wiki][HSTS wiki]
+* [HTTP Strict Transport Security Cheat Sheet][HSTS cheat sheat]
+
+## What does the rule check?
+
+For a site served over HTTPS, this rule checks the following:
+
+* If it has a `Strict-Transport-Security` header
+* If the header has the required `max-age` directive
+* If the `max-age` directive has a value that is longer than 18 weeks(10886400s)
+* If `Strict-Transport-Security` header has repetitive directives
+
+### Examples that **trigger** the rule
+
+```text
```

Comment:
Add short explanation for each example on why it triggers the rule (here is an [example](https://github.com/sonarwhal/sonar/blob/b013bc5dbf43e42acac34a4f6e621583bb0679a6/docs/user-guide/rules/content-type.md#examples-that-trigger-the-rule) on how it should look like).